providers/prometheus/interceptors.go

```diff
@@ -0,0 +1,133 @@
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"google.golang.org/grpc"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors"
+)
+
+type reporter struct {
+	clientMetrics   *ClientMetrics
+	serverMetrics   *ServerMetrics
+	typ             interceptors.GRPCType
+	service, method string
+	startTime       time.Time
+	kind            Kind
+}
+
+func (r *reporter) StartTimeCall(startTime time.Time, callType string) interceptors.Timer {
```

Comment:
I think I actually remove this as part of core reporter interface as we can totally put this logic into `reporter` function (below) logic, no? 

Look on `logging` interceptor it has exactly the same mechanism to report latency. (:

Reply:
Okay, I now get your point. Basically use the duration parameter of the reporter functions to get the job done. Nice optimization suggestion!